src/racer/nameres.rs

```diff
@@ -2488,8 +2500,70 @@ pub(crate) fn get_index_output(selfm: &Match, session: &Session) -> Option<Ty> {
         .next()
         .and_then(|(_, item_ty)| match item_ty {
             Ty::PathSearch(paths) => {
-                get_assoc_type_from_header(&paths.path, selfm, &index_header, session)
+                get_assoc_type_from_header(&paths.path, context, impl_header, session)
             }
             _ => Some(item_ty),
         })
 }
+
+/// Checks if trait_impl is the impl TraitName<OtherType>
+fn has_impl_for_other_type(
+    trait_impl: &ImplHeader,
+    trait_name: &str,
+    other_type: Option<&str>,
+) -> bool {
+    if let Some(ref path) = trait_impl.trait_path() {
+        if path.name() == Some(trait_name) {
+            if other_type.is_none() && path.segments[0].generics.len() == 0 {
+                return true;
+            }
+            if path.segments[0].generics.len() > 0 {
```

Comment:
What does it mean?
Is it for cases like `R = St<String>` and `Output = St<T>`?
Since trait_name is just a path name and doesn't include generics, I think it's impossible to judge if the R and Output's type parameter matches.

Reply:
No. It is for cases like `R = String` and `impl Add<String>...` I am trying to match `String` here. I should also recursively check generics for types like `R = MyType<String>` and `impl Add<MyType<String>> for ...` which I obviously missed. I will update the PR. Thanks.

Reply:
Ah I understand.
But I think `generics[0]` can be other than PathSearch(e.g. `&Point`).
Maybe we need some utility method for type matching?